src/wallet/wallet.ts

```diff
@@ -147,6 +160,25 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     await handleCreate(data.status);
   }
 
+  async onNewTx(newTx: any) {
+    const outputs: any[] = newTx.outputs;
+    let shouldGetNewAddresses = false;
+
+    for (const output of outputs) {
+      if (this.newAddresses.find((newAddress) => newAddress.address === output.decoded.address)) {
+        // break early
+        shouldGetNewAddresses = true;
+        break;
+      }
+    }
+
+    if (shouldGetNewAddresses) {
+      await this.getNewAddresses();
```

Comment:
I think it's important leave a comment here explaining why you are doing this. This is related to this TODO in the constructor, right? If so, we need to explain here the reason we are doing this and remove the TODO

```
    // TODO When we integrate the real time events from the wallet service
    // we will need to have a trigger to update this array every new transaction
    // because the new tx might have used one of those addresses
    // so we just need to call await this.getNewAddresses(); on the new tx event
```

Reply:
You are absolutely right, done in 11ca6458aaa1643201732a1f74e54da6f6bb9348